Optimize testing skill for input token consumption (#1667)
open-edge-platform/scenescape: .github/skills/testing/references/conftest-and-zephyr.md

<!--
SPDX-FileCopyrightText: (C) 2026 Intel Corporation
SPDX-License-Identifier: Apache-2.0
-->

# Conftest Patterns and Zephyr IDs

## Root `tests/conftest.py`

Owns session lifecycle:

- Session fixtures: `repo_root`, `version`, `secrets_dir`, `supass`, compose manager

> - Function fixtures: `scenescape_env`, `params`

Reviewer: ```suggestion
- Function fixtures: `scenescape_env`, `params`, `result_recorder`
```

- Hooks: profile-ordered collection, CLI options, per-test logging / container log collection

> - Records results using module `TEST_NAME` when present

Reviewer: ```suggestion
- Zephyr ID handling:
  - `@pytest.mark.test_name("NEX-T#####")` sets the XML test name
  - `result_recorder` prints `NEX-T#####: PASS` or `NEX-T#####: FAIL` on teardown
  - If `result_recorder.success()` is never reached, the result defaults to `FAIL`
```

## Functional `tests/functional/conftest.py`

Adds functional helpers (`rest`, `scene_uid`, etc.), `--env-profiles` matrix via `pytest_generate_tests`, and related setup fixtures.

## Unit-test Zephyr hooks (when using a local conftest)

> All tests need a Zephyr ID (`NEX-T#####`). Functional/UI tests typically set it via `record_xml_attribute` or `TEST_NAME` + env matrix. For unit suites that still use session hooks:

Reviewer: ```suggestion
All tests need a Zephyr ID (`NEX-T#####`). Functional/UI tests typically set it via @pytest.mark.test_name("NEX-T#####"). For unit suites that still use session hooks:
```
